src/sentry/rules/history/preview.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any, Dict, Sequence
+
+from django.utils import timezone
+
+from sentry.models import Project
+from sentry.rules import rules
+from sentry.rules.history.base import TimeSeriesValue
+
+PREVIEW_TIME_RANGE = timedelta(weeks=2)
+# limit on number of ConditionActivity's a condition will return
+# if limited, conditions should return the latest activity
+CONDITION_ACTIVITY_LIMIT = 1000
+
+
+def preview(
+    project: Project,
+    conditions: Sequence[Dict[str, Any]],
+    filters: Sequence[Dict[str, Any]],
+    condition_match: str,
+    filter_match: str,
+    frequency_minutes: int,
+) -> Sequence[TimeSeriesValue] | None:
+    end = timezone.now()
+    start = end - PREVIEW_TIME_RANGE
+    hours = get_hourly_bucket(PREVIEW_TIME_RANGE)
+    hourly_buckets = [0] * hours
+
+    # must have at least one condition to filter activity. Filters currently not supported
+    if len(conditions) == 0 or len(filters):
+        return None
+    # all the currently supported conditions are mutually exclusive
+    elif len(conditions) > 1 and condition_match == "all":
+        return [TimeSeriesValue(start + timedelta(hours=i), 0) for i in range(hours)]
+
+    activity_set = set()
+    for condition in conditions:
+        condition_cls = rules.get(condition["id"])
+        if condition_cls is None:
+            return None
+        # instantiates a EventCondition subclass and retrieves activities related to it
+        condition_inst = condition_cls(project, data=condition)
+        try:
+            activity_set.update(condition_inst.get_activity(start, end, CONDITION_ACTIVITY_LIMIT))
```

Comment:
So when we de-dupe here via the set, i assume it just compares whether the objects are identical. Probably fine for now, but is there a chance that in the future the same group might be in here multiple times (and does that matter?)

Reply:
i thought the same type of activity was created for both regression and reappearances, i've just tested it out again and activity objects aren't created for reappearances at all. It's probably better to have separate activity for these anyways. this set isn't needed